pkg/controller/mpi_job_controller.go

```diff
@@ -555,10 +546,11 @@ func (c *MPIJobController) syncHandler(key string) error {
 				return err
 			}
 		}
-
-		worker, err = c.getOrCreateWorker(mpiJob)
-		if err != nil {
-			return err
+		if !isMPIJobSuspended(mpiJob) {
+			worker, err = c.getOrCreateWorker(mpiJob)
```

Comment:
also create the podgroup conditionally?

Reply:
Now I create all resources for as long as the MPIJob is suspended. This is the common workflow for Kueue, where the Job is created suspended. In some cases, when the job never get unsuspended (for whatever reason), we can save on creating the objects.

Reply:
For now reverted the change - sorry for back end forth - this is just to minimize the diff as the unit tests currently verify the objects are created. Let me know what you think.